telos-backend/test/integration/calendar_api.test.js

```diff
@@ -0,0 +1,137 @@
+const { MongoMemoryServer } = require('mongodb-memory-server');
+const express = require('express');
+const mongoose = require('mongoose');
+const axios = require('axios');
+const routes = require('../../src/routes');
+const CalendarEvent = require('../../src/models/calendar_event');
+
+let mongod;
+let app;
+let server;
+let event1;
+let event2;
+let event3;
+
+beforeAll(async (done) => {
+  mongod = new MongoMemoryServer();
+
+  const connectionString = await mongod.getUri();
+  await mongoose.connect(connectionString, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    useFindAndModify: true,
+  });
+
+  app = express();
+  app.use(express.json());
+  app.use('/', routes);
+  server = app.listen(3000, () => done());
+});
+
+beforeEach(async () => {
+  const coll = await mongoose.connection.db.createCollection('calendarevents');
+
+  event1 = {
+    name: 'test1',
+    startTime: '2021-01-01T00:00:00',
+    endTime: '2021-01-01T05:00:00',
+  };
+
+  event2 = {
+    name: 'test2',
+    startTime: '2021-01-01T00:00:00',
+    endTime: '2021-01-01T05:00:00',
+  };
+
+  event3 = {
+    name: 'test3',
+    startTime: '2021-01-01T00:00:00',
+    endTime: '2021-01-01T05:00:00',
+  };
+
+  await coll.insertMany([event1, event2, event3]);
+});
+
+afterEach(async () => {
+  await mongoose.connection.db.dropCollection('calendarevents');
+});
+
+afterAll((done) => {
+  server.close(async () => {
+    await mongoose.disconnect();
+    await mongod.stop();
+
+    done();
+  });
+});
+
+it('Can post a calendar event', async () => {
+  const body = {
+    name: 'Test calendar',
+    startTime: '2021-01-01T02:00:00',
+    endTime: '2021-01-01T05:00:00',
+  };
+  const response = await axios.post('http://localhost:3000/api/calendar', {
+    name: 'Test calendar',
```

Comment:
Should this just be using the body JSON defined above instead of creating it again?

Reply:
Good point, removed.